tests/test_operations.py

```diff
@@ -0,0 +1,173 @@
+    dimensions_2 = ['time', 'var_name', 'tile_name']
+    dc_reader_2 = DataCubeReader.from_filepaths(gt_filepaths[1:2], YeodaFilename, dimensions=dimensions_2,
+                                                stack_dimension='time', tile_dimension='tile_name')
+    dc_reader_intsct = dc_reader_1.intersect(dc_reader_2)
+    assert len(dc_reader_intsct) == 2
+    assert 'var_name' not in dc_reader_intsct.dimensions
+    assert 'band' not in dc_reader_intsct.dimensions
+    assert 'time' in dc_reader_intsct.dimensions
+
+
+def test_align_dimension_intersect_shrink(gt_filepaths):
+    dimensions = ['time', 'tile_name']
+    dc_reader_1 = DataCubeReader.from_filepaths(gt_filepaths, YeodaFilename, dimensions=dimensions,
+                                                stack_dimension='time', tile_dimension='tile_name')
+    dc_reader_2 = dc_reader_1.clone()
+    timestamps = list(set(dc_reader_1['time']))
+    timestamps_sel = timestamps[2:]
+    dc_reader_2.select_by_dimension(lambda t: t.isin(timestamps_sel), name='time', inplace=True)
+
+    dc_reader_aligned = dc_reader_1.align_dimension(dc_reader_2, name='time')
+    dc_reader_intersected = dc_reader_1.intersect(dc_reader_2, on_dimension='time')
+
+    assert sorted(list(dc_reader_aligned['time'])) == sorted(list(dc_reader_intersected['time']))
+
+
+def test_align_dimension_shrink(gt_filepaths):
+    dimensions = ['time', 'tile_name']
+    dc_reader_1 = DataCubeReader.from_filepaths(gt_filepaths, YeodaFilename, dimensions=dimensions,
+                                                stack_dimension='time', tile_dimension='tile_name')
+    dc_reader_2 = dc_reader_1.clone()
+    timestamps = sorted(list(set(dc_reader_1['time'])))
+    timestamps_sel = timestamps[2:]
+    dc_reader_2.select_by_dimension(lambda t: t.isin(timestamps_sel), name='time', inplace=True)
+
+    dc_reader_1.align_dimension(dc_reader_2, name='time', inplace=True)
+    assert sorted(list(set(dc_reader_1['time']))) == timestamps_sel
+
+
+def test_align_dimension_grow(gt_filepaths):
+    dimensions = ['time', 'tile_name']
+    dc_reader_1 = DataCubeReader.from_filepaths(gt_filepaths, YeodaFilename, dimensions=dimensions,
+                                                stack_dimension='time', tile_dimension='tile_name')
+    dc_reader_2 = dc_reader_1.clone()
+    timestamps = sorted(list(set(dc_reader_1['time'])))
+    timestamps_sel = timestamps[:1]
+    dc_reader_2.select_by_dimension(lambda t: t.isin(timestamps_sel), name='time', inplace=True)
+    tile_names = sorted(list(set(dc_reader_1['tile_name'])))
+    tile_names_sel = tile_names[:1]
+    dc_reader_2.select_by_dimension(lambda t: t.isin(tile_names_sel), name='tile_name', inplace=True)
+
+    dc_reader_2.align_dimension(dc_reader_1, name='time', inplace=True)
+    dc_reader_1.select_by_dimension(lambda t: t.isin(timestamps_sel), name='time', inplace=True)
+    assert len(dc_reader_1) == len(dc_reader_2)
```

Comment:
those are some nice concise and readable tests, well done :)

Reply:
Thank you!